src/FindIntrinsics.cpp

```diff
@@ -493,6 +494,25 @@ class FindIntrinsics : public IRMutator {
                         rounding_mul_shift_right(x, y, cast(unsigned_type, c0)),
                         is_x_same_int_or_uint && c0 >= bits) ||
 
+                // We can also match on smaller shifts if one of the args is
+                // narrow. We don't do this for signed (yet), because the
+                // saturation issue is tricky.
```

Comment:
Do we think we will ever do so, tricky or not? (If so, tracking issue, please)

Reply:
Not sure. It depends on if it comes up in practice. It's part of the unbounded frontier of ways we could make instruction selection more clever. I don't want to add a tracking issue for each possible new way to use an instruction.